tests/engines/collections/acceptance/moderation/moderators-remove-self-test.ts

```diff
@@ -0,0 +1,58 @@
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { percySnapshot } from 'ember-percy';
+import { module, test } from 'qunit';
+
+import { click, visit } from 'ember-osf-web/tests/helpers';
+import { setupEngineApplicationTest } from 'ember-osf-web/tests/helpers/engines';
+import { currentRouteName } from '@ember/test-helpers/setup-application-context';
+
+module('Collections | Acceptance | moderation | moderators | remove self', hooks => {
+    setupEngineApplicationTest(hooks, 'collections');
+    setupMirage(hooks);
+
+    test('it removes a self collection moderator', async assert => {
+        // Given I create a collection, collection-provider and a second moderator
+        server.create('user', 'loggedIn');
+        const primaryCollection = server.create('collection');
+        const provider = server.create('collection-provider', {
+            id: 'studyswap',
+            primaryCollection,
+        }, 'currentUserIsModerator');
+        server.createList('moderator', 1, { provider });
+
+        // When I visit the discover page
+        await visit(`/collections/${provider.id}/discover`);
+
+        // When I click on the moderation tab
+        await click('[data-test-branded-navbar-moderation]');
+
+        // When I click on the moderators tab
+        await click('[data-test-collections-moderation-moderators-tab]');
+
+        // Then I assert both moderators exist
+        assert.dom('[data-test-moderator-row]').exists({ count: 2 }, '1 moderator is listed');
+        // Then I assert that only I have a delete button as a moderator and not admin
+        assert.dom('[data-test-delete-button]').exists({ count: 1 },
+            'Can delete a single moderator, including self');
+
+        // When I click on the delete button as myself
+        await click('[data-test-delete-button]');
+        // When I delete, I have to update the provider permissions manually to
+        // simulate the server response
+        provider.update({permissions: []});
+
+        // When I click on the confirm remove moderator dialog
+        await click('[data-test-confirm-delete]');
+
+        // Then I verify the moderation button is no longer visable
+        assert.dom('[data-test-branded-navbar-moderation]').doesNotExist('The moderation button does not exist.');
+
+        // When I attempt to go to the moderators page from the url
+        await visit(`/collections/${provider.id}/moderation/moderators`);
```

Comment:
I think asserting that users get rerouted to the discover page after removing themselves as a moderator would be useful here too, just to document what the expected behavior should be